### DISCUSSION TITLE
Persistent Jump Locations / Node Choices discussion

### DISCUSSION THREAD

**Discussion by ziontee113** [L3MON4D3/LuaSnip#394] [Ideas]
Hi @L3MON4D3, I'm been thinking of this topic for a while. Could we see in the future, LuaSnip with Persistent Jump Locations & Node Choices?

Writing code with LuaSnip is a joy. I hope maybe we can see LuaSnip being used when we refactor our code as well?

Is this feature easy or difficult to implement in LuaSnip? I'd love to hear your thoughts about this topic!

**Comment by L3MON4D3** [comment]
Persistent like across neovim-session?
That would be really cool, afaict we'd need to serialize all the snippets and extmarks(!!), not sure if there already exists something that handles all of our requirements (metatables and self-referencing tables).
Applying the snippet structure to existing text should also be possible.
Very intriguing, but complicated.
